### PR TITLE
Replace XHRSubmitFormController with TurboFormController for declarative form submission hijack

### PR DESCRIPTION
**Pull request by pkong-ds**
<details>
<summary>Outdated</summary>

## Motivation
As per offline discussion, it was found that `<form>` `submit` events using stimulius syntax like below did not stop other methods the form from submitting

```html
<!-- html -->
<form
...
data-action="submit->someController#someMethod"
>
...
```

```ts
// someController.ts

class SomeController extends Controller {
  someMethod(e: Event) {
    e.preventDefault()
    e.stopImmediatePropagation() // This did not stop other actions
  }
}

```

This PR exists to make `e.stopImmediatePropagation` in `someMethod` works as expected

</details>

## Regression Test

Tested turbo-enabled forms submission, still behaves like SPA

## How are you sure all forms are updated?

Total 36 forms

```bash
rg '<form' ./resources/authgear/templates/en/web/authflowv2 | rg '<form' -c
```

30 forms have `data-controller="turbo-form"` added

```bash
rg '<form' ./resources/authgear/templates/en/web/authflowv2 -C 10 | rg 'data-controller="turbo-form"' -c
```

6 forms were originally turbo-disabled form, added comment to specify

```bash
rg '<form' ./resources/authgear/templates/en/web/authflowv2 -C 5 | rg 'Form with disabled' -c
```

### CODE REVIEW

**Comment by pkong-ds** [comment]
@louischan-oursky 🙏 Can see if this works? Will based on this change to implement bot protection

**Comment by pkong-ds** [comment]
@louischan-oursky updated as per offline discussion, thanks!

**Review by louischan-oursky — COMMENTED** [review]
1. signup.html -> Remove data-turbo=false in button, do not attach the controller, keep data-turbo=false in form
2. login.html -> For the login ID form, add data-turbo=false to the form, remove data-turbo=false in the button. Move the comment to the data-turbo=false of the form. For oauth form, same as 1
3. select_account.html -> move data-turbo=false from the button to the form, do not attach the controller
4. terminate_other_sessions.html -> same as 3
5. view_recovery_code.html -> same as 3

**Comment by pkong-ds** [comment]
Updated, Thanks!

## Preview
### `signup.html`
https://github.com/user-attachments/assets/ec3983f6-98f7-4931-a3a7-b6d9dfb063b2

### `login.html`

https://github.com/user-attachments/assets/cba45306-9fca-4ada-9e3d-c2c52d0fb897

File: resources/authgear/templates/en/web/authflowv2/oob_otp_link.html [hunk 1]
```diff
@@ -73,14 +80,20 @@ <h1 class="screen-title">
...
+  >
     {{ $.CSRFField }}
     <button
       class="primary-btn w-full"
       type="submit"
       name="x_action"
       value="resend"
-      data-controller="countdown"
+      data-controller="countdown turbo-form"
+      data-action="submit->turbo-form#submitForm"
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/web/authflowv2/oob_otp_link.html]: This is redundant.

File: resources/authgear/templates/en/web/authflowv2/terminate_other_sessions.html [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {{ template "authflowv2/__page_frame.html" . }}
 
 {{ define "page-content" }}
+  {{/* Form with disabled turbo drive, submit button has data-turbo="false" */}}
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/web/authflowv2/terminate_other_sessions.html]: Move data-turbo=false here

**Comment by pkong-ds** [comment]
e2e failed :sadfrog: can help retry?

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**